Orange/base.py

```diff
@@ -320,9 +321,12 @@ def data_to_model_domain():
 
             if self.original_domain.attributes != data.domain.attributes \
                     and data.X.size \
-                    and not np.isnan(data.X).all():
+                    and not np.isnan(
```

Comment:
I don't think this is right. You want to check if all the entries are not nan, but then you only check the non-zero entries for sparse matrices. This behaviour is not consistent with what will happen for dense matrices. Also, consider taking a look at potentially useful sparse utilities [here](https://github.com/biolab/orange3/blob/master/Orange/statistics/util.py#L623-L659) and use those where possible. If a use case is missing, such as the one you're introducing right now, please add it there, otherwise, there's a lot of duplication.

Reply:
You are right. :) I totally forgot that in sparse matrix unknown values are actually zeroes. Now it is fixed. I implemented that as a function similar to any_nan.